unblob/handlers/archive/instar/instar_hd.py

```diff
@@ -0,0 +1,44 @@
+from pathlib import Path
+
+from structlog import get_logger
+
+from unblob.file_utils import File, iterate_file
+from unblob.handlers.archive.zip import ZIPHandler
+from unblob.models import Extractor, HexString
+
+logger = get_logger()
+
+LOCAL_FILE_HEADER_ZIP = b"\x50\x4b\x03\x04"
+LOCAL_FILE_HEADER_INSTAR = b"\x50\x4b\x03\x07"
+
+EOCD_ZIP = b"\x50\x4b\x05\x06"
+EOCD_INSTAR = b"\x50\x4b\x05\x09"
+
+CD_FILE_HEADER_ZIP = b"\x50\x4b\x01\x02"
+CD_FILE_HEADER_INSTAR = b"\x50\x4b\x01\x08"
+
+
+class InstarHDExtractor(Extractor):
+    def fix_headers(self, chunk: bytes) -> bytes:
+        return (
+            chunk.replace(LOCAL_FILE_HEADER_INSTAR, LOCAL_FILE_HEADER_ZIP)
+            .replace(EOCD_INSTAR, EOCD_ZIP)
+            .replace(CD_FILE_HEADER_INSTAR, CD_FILE_HEADER_ZIP)
```

Comment:
Using `replace` can also cause some files not extracting: if the patterns occur in the content, not in the header - small probability, but can be triggered intentionally. E.g. zip files can store files without compression, it is sometimes useful to embed a zip file in a zip file, in this case the internal zip/INSTAR would be extracted with a crc error.

The proper solution would be to walk the zip structures starting from the EOCD record and using the central directory records to locate the local headers, and fix the magic numbers for them one by one.